zerver/views/portico.py

```diff
@@ -36,6 +37,9 @@ def plans_view(request: HttpRequest) -> HttpResponse:
     realm_plan_type = 0
     free_trial_days = settings.FREE_TRIAL_DAYS
     sponsorship_pending = False
+    sponsorship_url = "/upgrade#sponsorship"
+    if is_subdomain_root_or_alias(request):
+        sponsorship_url = "/accounts/go/?next=/upgrade%23sponsorship"
```

Comment:
Don't hand-code URL-quoting; there are functions for this that let us avoid repeating ourselves too:

```
diff --git a/zerver/views/portico.py b/zerver/views/portico.py
index 33cc6f456f..8d2726bd9b 100644
--- a/zerver/views/portico.py
+++ b/zerver/views/portico.py
@@ -1,4 +1,5 @@
 from typing import Optional
+import urllib.parse
 
 import orjson
 from django.conf import settings
@@ -39,7 +40,8 @@ def plans_view(request: HttpRequest) -> HttpResponse:
     sponsorship_pending = False
     sponsorship_url = "/upgrade#sponsorship"
     if is_subdomain_root_or_alias(request):
-        sponsorship_url = "/accounts/go/?next=/upgrade%23sponsorship"
+        # If we're on the root domain, we make this link first ask you which organization.
+        sponsorship_url = f"/accounts/go/?next={urllib.parse.quote(sponsorship_url)}"
 
     if realm is not None:
         realm_plan_type = realm.plan_type
```

(And then it's great that that the test verifies what this actually produces)

Reply:
That's my bad. I will keep that in mind from now on! Thanks for merging! :)